pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook.go

```diff
@@ -0,0 +1,85 @@
+package fleetresourcehandler
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"go.goms.io/fleet/pkg/webhook/validation"
+)
+
+const (
+	// ValidationPath is the webhook service path which admission requests are routed to for validating custom resource definition resources.
+	ValidationPath = "/validate-v1-fleetresourcehandler"
+	groupMatch     = `^[^.]*\.(.*)`
+)
+
+// Add registers the webhook for K8s bulit-in object types.
+func Add(mgr manager.Manager) error {
+	hookServer := mgr.GetWebhookServer()
+	hookServer.Register(ValidationPath, &webhook.Admission{Handler: &fleetResourceValidator{Client: mgr.GetClient()}})
+	return nil
+}
+
+type fleetResourceValidator struct {
+	Client  client.Client
+	decoder *admission.Decoder
+}
+
+func (v *fleetResourceValidator) Handle(_ context.Context, req admission.Request) admission.Response {
+	var response admission.Response
+	if req.Operation == admissionv1.Create || req.Operation == admissionv1.Update || req.Operation == admissionv1.Delete {
+		switch req.Kind.String() {
+		case retrieveCRDGVK():
+			klog.V(2).InfoS("handling CRD resource", "GVK", retrieveCRDGVK())
+			response = v.handleCRD(req)
+		default:
+			klog.V(2).InfoS("resource is not monitored by fleet resource validator webhook", "GVK", req.Kind.String())
+			response = admission.Allowed("")
+		}
+	}
+	return response
+}
+
+func (v *fleetResourceValidator) handleCRD(req admission.Request) admission.Response {
+	var crd v1.CustomResourceDefinition
+	if req.Operation == admissionv1.Delete {
+		// req.Object is not populated for delete: https://github.com/kubernetes-sigs/controller-runtime/issues/1762.
+		if err := v.decoder.DecodeRaw(req.OldObject, &crd); err != nil {
+			klog.ErrorS(err, "failed to decode old request object for delete operation", "userName", req.UserInfo.Username, "groups", req.UserInfo.Groups)
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+	} else {
+		if err := v.decoder.Decode(req, &crd); err != nil {
+			klog.ErrorS(err, "failed to decode request object for create/update operation", "userName", req.UserInfo.Username, "groups", req.UserInfo.Groups)
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+	}
+
+	group := regexp.MustCompile(groupMatch).FindStringSubmatch(crd.Name)[1]
```

Comment:
Hi Arvind! Apologies if I misunderstand anything here, but the intention here was to match against GVKs (specifically groups), right? But the code is now matching against object names.

Reply:
Of course it's not wrong considering that the name always follows the same pattern, but we do have direct access to GVK in admission request.

Reply:
Hi Arvind! For this one, is there any concern over checking the GVK directly?

Reply:
The GVK and GVR contained in the request contains the info fro CustomResourceDefintiion itself

Reply:
Hi Arvind! In this case, would you mind adding a comment here to explain the reasoning, specifically the naming pattern CRD uses? Thanks!